src/RequestExecutor/Request.spec.ts

```diff
@@ -0,0 +1,92 @@
+import 'reflect-metadata';
+import type { Request as RequestInterface } from './Request';
+
+describe('Request', () => {
+  let readFileMock!: jest.Mock;
+  let Request!: typeof RequestInterface;
+  const noError: NodeJS.ErrnoException | null = null;
+  const certContent = Buffer.from([]);
+  beforeEach(async () => {
+    readFileMock = jest.fn();
+    jest.doMock('fs', () => ({ readFile: readFileMock }));
+    // ADHOC: `jest.doMock` must be called before importing SUT
+    Request = (await import('./Request')).Request;
+  });
+
+  afterEach(() => {
+    jest.resetModules();
+    readFileMock.mockReset();
+  });
+
```

Comment:
Please add nested `describe` block for method
```suggestion
  describe('setCerts', () => {)
```